### PR TITLE
Implement cat.transform

### PR DESCRIPTION
**Pull request by codebrain**
Implement: https://www.elastic.co/guide/en/elasticsearch/reference/master/cat-transforms.html

### CODE REVIEW

File: tests/Tests/Cat/CatTransform/CatTransformApiTests.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+using FluentAssertions;
+using Nest;
+using Tests.Core.Extensions;
+using Tests.Core.ManagedElasticsearch.Clusters;
+using Tests.Framework.EndpointTests;
+using Tests.Framework.EndpointTests.TestState;
+
+namespace Tests.Cat.CatTransform
+{
+	public class CatTransformApiTests
```
Comment (Mpdreamz) [inline tests/Tests/Cat/CatTransform/CatTransformApiTests.cs]: Needs `SkipVersion` attributes:

https://dev.azure.com/azure3/elasticsearch-net/_build/results?buildId=3445&view=logs&j=887c6f23-4c3d-5668-6783-9f77056b8369&t=231ee5e6-21f5-558b-5c1b-be75406f6d88&l=3534

Comment (codebrain) [reply tests/Tests/Cat/CatTransform/CatTransformApiTests.cs]: ++

**Comment by codebrain** [comment]
`CatTransformRecord` needs updating: https://www.elastic.co/guide/en/elasticsearch/reference/7.7/cat-transforms.html

**Comment by codebrain** [comment]
Closed in favor of https://github.com/elastic/elasticsearch-net/pull/4557

**Pull request closed without merging**